typescript/infra/src/govern/HyperlaneAppGovernor.ts

```
@@ -120,32 +135,92 @@ export abstract class HyperlaneAppGovernor<
+      safeOwner = this.checker.configMap[chain].owner as Address;
+    } else {
+      safeOwner = (this.checker.configMap[chain].owner as AccountConfig).owner;
+    }
     await sendCallsForType(
       SubmissionType.SAFE,
-      new SafeMultiSend(this.checker.multiProvider, chain, owner),
+      new SafeMultiSend(this.checker.multiProvider, chain, safeOwner),
     );
     await sendCallsForType(SubmissionType.MANUAL, new ManualMultiSend(chain));
   }
 
   protected pushCall(chain: ChainName, call: AnnotatedCallData) {
+    this.calls[chain] = this.calls[chain] || [];
     this.calls[chain].push(call);
   }
 
+  protected popCall(chain: ChainName): AnnotatedCallData | undefined {
+    return this.calls[chain].pop();
+  }
+
   protected abstract mapViolationsToCalls(): Promise<void>;
 
   protected async inferCallSubmissionTypes() {
     for (const chain of Object.keys(this.calls)) {
       for (const call of this.calls[chain]) {
-        call.submissionType = await this.inferCallSubmissionType(chain, call);
+        let submissionType = await this.inferCallSubmissionType(chain, call);
+        if (!submissionType) {
+          submissionType = await this.inferICAEncodedSubmissionType(
+            chain,
+            call,
+          );
+        }
+        call.submissionType = submissionType;
       }
     }
   }
 
+  protected async inferICAEncodedSubmissionType(
+    chain: ChainName,
+    call: AnnotatedCallData,
+  ): Promise<SubmissionType> {
+    const multiProvider = this.checker.multiProvider;
+    const signer = multiProvider.getSigner(chain);
+    if (this.interchainAccount) {
+      const ownableAddress = call.to;
+      const ownable = Ownable__factory.connect(ownableAddress, signer); // mailbox
+      const account = Ownable__factory.connect(await ownable.owner(), signer);
+      const localOwner = await account.owner();
+      if (eqAddress(localOwner, this.interchainAccount.routerAddress(chain))) {
```

maybe throw an error if this is not true? or log

---

logged